server/api/locations/locations.api.js

```diff
@@ -15,54 +29,129 @@ exports.create = route(async (req, res) => {
   res.status(201).send(await serialize(req, location, policy));
 });
 
+/**
+ * Lists locations ordered by name.
+ *
+ * @function
+ */
+exports.list = route(async (req, res) => {
+
+  const query = new Location();
+  const locations = await query.orderBy('name').orderBy('created_at').fetchAll();
```

Comment:
I have trouble understanding this line.
`Location` refers to the model, I thus supposed that doing `new Location()` would return a new Location.
How can you do two `orderBy`s on this new Location and then a `fetchAll()` to get all `Locations` ?
I'm probably missing something.

Reply:
It's because the Bookshelf ORM is a bit weird: creating records is how you make queries.

```js
await new Location({ name: 'foo' }).fetch() // => SELECT * FROM locations WHERE name = 'foo';
```

It can also be written this way:

```js
await new Location().where('name', 'foo').fetch() // => SELECT * FROM locations WHERE name = 'foo';
```

So basically, any record is also a query builder.

Reply:
Okay.
Hence the `const query`.